examples/nextjs-with-typescript/src/Link.tsx

```diff
@@ -1,36 +1,42 @@
 /* eslint-disable jsx-a11y/anchor-has-content */
-import React from 'react';
+import * as React from 'react';
 import clsx from 'clsx';
 import { useRouter } from 'next/router';
-import NextLink from 'next/link';
-import MuiLink, { LinkProps as MuiLinkProps } from '@material-ui/core/Link';
+import NextLink, { LinkProps as NextLinkProps } from 'next/link';
+import MuiLink from '@material-ui/core/Link';
+import { TypographyProps } from '@material-ui/core/Typography';
 
-interface NextLinkProps {
-  as?: string;
-  href?: string;
-  prefetch?: boolean;
+interface NextLinkExtendedProps extends Omit<NextLinkProps, 'href' | 'onError'> {
+  className?: string;
+  href: string;
 }
 
-type NextComposedProps = React.AnchorHTMLAttributes<HTMLAnchorElement> & NextLinkProps;
-
-const NextComposed = React.forwardRef(function NextComposed(
-  props: NextComposedProps,
-  ref: React.Ref<any>,
-) {
-  const { as, href, prefetch, ...other } = props;
+const NextComposed = React.forwardRef<HTMLAnchorElement, NextLinkExtendedProps>((props, ref) => {
+  const { as, href, replace, scroll, passHref, shallow, prefetch, ...other } = props;
 
   return (
-    <NextLink href={href} prefetch={prefetch} as={as}>
+    <NextLink
+      href={href}
+      prefetch={prefetch}
+      as={as}
+      replace={replace}
+      scroll={scroll}
+      shallow={shallow}
+      passHref={passHref}
+    >
       <a ref={ref} {...other} />
     </NextLink>
   );
 });
 
-interface LinkProps extends MuiLinkProps, NextLinkProps {
+type LinkProps = LinkPropsBase & Pick<TypographyProps, 'align' | 'color' | 'display' | 'variant'>;
```

Comment:
What about the other props of the Link component? variantMapping, noWrap, gutterBottom, underline?

Reply:
Added all of the missing properties. Not sure if there can be a better solution to `& {underline?: 'none' | 'hover' | 'always'};` 

Should there be a exported prop type for Link that includes underline?

https://github.com/mui-org/material-ui/blob/b198da072707eb6da1b5579b35eba989e26a6be4/packages/material-ui/src/Link/Link.d.ts#L9

Reply:
I have tried to make an alternative implementation leveraging LinkProps without any success. Would using this type be more future proof?

Reply:
Yeah, just to avoid defining it again. I guess it is what it is at this time.